docs/virtual_machines/run_strategies.md

```diff
@@ -18,22 +18,26 @@ has been introduced. This is mutually exclusive with `Running` as they
 have somewhat overlapping conditions. There are currently four
 RunStrategies defined:
 
--   Always: A VirtualMachineInstance will always be present. If the
-    VirtualMachineInstance crashed, a new one will be spawned. This is
-    the same behavior as `spec.running: true`.
-
--   RerunOnFailure: A VirtualMachineInstance will be respawned if the
-    previous instance failed in an error state. It will not be
-    re-created if the guest stopped successfully (e.g. shut down from
-    inside guest).
-
--   Manual: The presence of a VirtualMachineInstance or lack thereof is
-    controlled exclusively by the start/stop/restart VirtualMachine
-    subresource endpoints.
-
--   Halted: No VirtualMachineInstance will be present. If a guest is
-    already running, it will be stopped. This is the same behavior as
-    `spec.running: false`.
+-   Always: The system is asked to always have the VM running.
+    This is achieved by respanwing a VirtualMachineInstance whenever
+    the current one terminated in a controlled (e.g. shutdown from
+    inside the guest) or uncontrolled (e.g. crash) way.
+    This behavior is equal to `spec.running: true`.
+
+-   RerunOnFailure: Similar to `Always`, except that the VM is only
+    restarted if it terminated in an uncontrolled way (e.g. crash).
```

Comment:
I have used this in the past, but I still don't really understand how the system determines whether the termination is controlled or uncontrolled. A note about this or a link to how qemu/libvirt determines this would be really helpful. (again, optional)

Reply:
Addressed.